crates/router/src/core/payments/operations/payment_confirm.rs

```diff
@@ -1102,19 +1105,84 @@ impl<F: Clone + Send + Sync> Domain<F, api::PaymentsRequest, PaymentData<F>> for
                 )
                 .await?;
 
-                payment_data.payment_method_data =
-                    network_token.map(domain::PaymentMethodData::NetworkToken);
+                let network_token = match response.response.clone() {
```

Comment:
Return status from the branch and have one                         unified_authentication_service::create_new_authentication(